docs/conceptual/More-about-how-ROCm-uses-PCIe-Atomics.rst

```diff
@@ -1,156 +1,63 @@
+`HSA Platform System Architecture Specification <http://hsafoundation.com/wp-content/uploads/2021/02/HSA-SysArch-1.2.pdf>`_.
 
-ROCm PCIe feature and overview of BAR memory
-================================================================
+ROCm uses platform atomics to perform memory operations like queuing, signaling, and synchronization across multiple CPU, GPU agents, and I/O devices. Platform atomics ensures that atomic operations run synchronously without interruptions or conflicts across multiple shared resources. Similarly, it enables memory alignment and synchronization of processing data within the shared memory locations and I/O devices.  This sections details how ROCm uses platform atomics and features of PCIe 3.0.
 
-ROCm is an extension of HSA platform architecture, so it shares the queuing model, memory model,
-signaling and synchronization protocols. Platform atomics are integral to perform queuing and
-signaling memory operations where there may be multiple-writers across CPU and GPU agents.
+Platform atomics in ROCm
+==============================
+Platform atomics enable the set of atomic operations that perform RMW actions across multiple processors, devices, and memory locations so that they run synchronously without interruption. An atomic operation is a sequence of computing instructions run as a single, indivisible unit. These instructions are completed in their entirety without any interruptions. If the instructions can't be completed as a unit without interruption, none of the instructions are run. These operations support 32-bit and 64-bit address formats.
 
-The full list of HSA system architecture platform requirements are here:
-`HSA Sys Arch Features <http://hsafoundation.com/wp-content/uploads/2021/02/HSA-SysArch-1.2.pdf>`_.
+ROCm uses platform atomics to:
 
-AMD ROCm Software uses the new PCI Express 3.0 (Peripheral Component Interconnect Express [PCIe]
-3.0) features for atomic read-modify-write transactions which extends inter-processor synchronization
-mechanisms to IO to support the defined set of HSA capabilities needed for queuing and signaling
-memory operations.
+* Update the HSA queue's ``read_dispatch_id``. The command processor on the GPU agent uses a 64-bit atomic add operation. It updates the packet ID it processed.
+* Update the HSA queue's ``write_dispatch_id``. The CPU and GPU agents use a 64-bit atomic add operation. It supports multi-writer queue insertions.
+* Update HSA Signals. A 64-bit atomic operation is used for CPU & GPU synchronization.
```

Comment:
This list is probably incomplete. It completely ignores the possibility that application shader programs can use atomic instructions which get translated into platform atomics if they target locations in system memory (at least for the supported subset of operations).

Reply:
Updated the intro text to present as some of the examples of operations, rather than the only operations in ROCm.